public/scripts/plugins/pcr/lib/sequence.js

```diff
@@ -27,7 +27,11 @@ class Sequence {
   }
 
   allFields () {
-    return this.requiredFields().concat(['id', 'name', 'antisense']);
+    return this.requiredFields().concat(this.optionalFields());
+  }
+
+  optionalFields () {
```

Comment:
Unless we can declare them `readonly` I prefer to put these as functions that return a fresh copy of an array each time.  Things can get accidentally edited and it becomes very confusing.  For example `slice` will edit an array associated as an attribute.  Just defensive programming.